rasa/nlu/components.py

```diff
@@ -87,13 +87,40 @@ def validate_required_components_from_data(
     pipeline: List["Component"], data: TrainingData
 ):
 
-    response_selector_exists = False
-    for component in pipeline:
-        # check if a response selector is part of NLU pipeline
-        if MESSAGE_RESPONSE_ATTRIBUTE in component.provides:
-            response_selector_exists = True
+    # check if there are regex features
+    if data.regex_features and not any(
+        [component.name == "RegexFeaturizer" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data with Regex's, but "
+            "haven't setup RegexFeaturizer in your config. "
+            "Either add RegexFeaturizer to your config or "
+            "remove the regex training data"
+        )
```

Comment:
For these (in comparison to the response selector), you don't necessarily have to do either, i.e. having regexes lying around when you don't want to use them won't break anything. So I'd rather make the second sentence something like `To featurize regexes, add a RegexFeaturizer to your pipeline.`

Reply:
gotcha makes sense have updated it accordingly

Reply:
Here by these, i meant regexes, lookup, and synonyms